### PR TITLE
Implement GraphQL

### PR DESCRIPTION
**Pull request by brambrink**

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -7,10 +7,11 @@
     "cypress": "cypress open",
     "cypress:run": "cypress run",
     "build": "next build",
-    "dev": "next & npm run json-server",
+    "dev": "next & npm run json-server & npm run json-graphql-server",
```
Comment (RubenWerdmuller) [inline package.json:10]: Rather than starting both the `REST` and `GraphQL` server, create separate commands for these

File: pages/_app.tsx [hunk 2]
```diff
@@ -0,0 +1,22 @@
+import { ApolloProvider } from "@apollo/client";
+import { AppProps } from "next/app";
+import { ThemeProvider } from "styled-components";
+import client from "../apollo-client";
```
Comment (RubenWerdmuller) [inline pages/_app.tsx:4]: Use named export for `client`

File: pages/index.tsx [hunk 3]
```diff
@@ -1,44 +1,33 @@
-import { ThemeProvider } from "styled-components";
+import client from "../apollo-client";
 
-import { StoreProvider } from "../src/providers";
-
-import GlobalStyle from "../styles/global";
-import { defaultTheme } from "../styles/theme";
+import { GET_DATA } from "../graphql/time-entries";
```
Comment (RubenWerdmuller) [inline pages/index.tsx:3]: Use more descriptive name for data

File: pages/index.tsx [hunk 4]
```diff
@@ -1,44 +1,33 @@
...
-      <GlobalStyle />
-      <ThemeProvider theme={defaultTheme}>
-        <Header />
-        <PageContainer>
-          <TimeEntries {...{ initialTimeEntries, clients }} />
-        </PageContainer>
-      </ThemeProvider>
-    </StoreProvider>
+    <>
+      <title> Time entries | Team Awesome Bram </title>
```
Comment (RubenWerdmuller) [inline pages/index.tsx:13]: Remove spacing

File: pages/team-members.tsx [hunk 5]
```diff
@@ -1,43 +1,32 @@
...
-      <GlobalStyle />
-      <ThemeProvider theme={defaultTheme}>
-        <Header />
-        <PageContainer>
-          <TeamMembers {...{ initialTeamMembers }} />
-        </PageContainer>
-      </ThemeProvider>
-    </StoreProvider>
+    <>
+      <title> Team members | Team Awesome Bram </title>
```
Comment (RubenWerdmuller) [inline pages/team-members.tsx:13]: Trim title

File: src/components/team-members/TeamMembers.tsx [hunk 6]
```diff
@@ -29,10 +34,18 @@ export const TeamMembers = ({ initialTeamMembers }: InitialTeamMembersProps) =>
...
+    addNewTeamMember({
+      variables: {
+        emailAddress: newTeamMember.emailAddress,
+        firstName: newTeamMember.firstName,
+        lastName: newTeamMember.lastName,
+        role: newTeamMember.role,
+        employer: newTeamMember.employer,
+        startingDate: newTeamMember.startingDate,
+      },
+    });
```
Comment (RubenWerdmuller) [inline src/components/team-members/TeamMembers.tsx:46]:
```suggestion
    const {emailAddress, firstName, lastName, role, employer, startingDate} = newTeamMember
    addNewTeamMember({
      variables: {emailAddress, firstName, lastName, role, employer, startingDate},
    });
```

**Comment by RubenWerdmuller** [comment]
[Very nice!](https://www.youtube.com/watch?v=LBduNcf1eQc&ab_channel=georgenajjar)

**Review by RubenWerdmuller — CHANGES_REQUESTED** [review]

File: pages/index.tsx (hunk 3 above)
Comment (brambrink) [reply pages/index.tsx:3]: This contains both the client- and time entries, I am struggling to find a better name for this, any suggestions?

File: src/components/team-members/TeamMembers.tsx (hunk 6 above)
Comment (brambrink) [reply src/components/team-members/TeamMembers.tsx:46]: This does not work this way with GraphQL I think. Right now, I am setting the state using: 
```
 const [addNewTeamMember] = useMutation(ADD_TEAM_MEMBER, {
    onCompleted: (data) => {
      setTeamMembers([...teamMembers, data.createTeamMember]);
    },
  });```

**Review by RubenWerdmuller — CHANGES_REQUESTED** [review]

File: pages/team-members.tsx [hunk 7]
```diff
@@ -1,16 +1,16 @@
-import client from "../apollo-client";
+import { client } from "../apollo-client";
 
 import { Header } from "../src/components/header/Header";
 import { PageContainer } from "../src/components/page-container";
 import { TeamMembers } from "../src/components/team-members";
 
 import { InitialTeamMembersProps } from "../src/types/Types";
-import { GET_TEAM_MEMBERS } from "../graphql/team-members";
+import { GET_TEAM_MEMBERS } from "../src/graphql/team-members";
```
Comment (RubenWerdmuller) [inline pages/team-members.tsx:8]: 👌!

File: pages/index.tsx (hunk 3 above)
Comment (brambrink) [reply pages/index.tsx]: Nice, thanks!

**Review by RubenWerdmuller — APPROVED** [review]
Nice 🔥🔥🔥🔥💧💧🧯🧑‍🚒🧑‍🚒 🚒

File: package.json [hunk 8]
```diff
@@ -7,10 +7,12 @@
     "cypress": "cypress open",
     "cypress:run": "cypress run",
     "build": "next build",
-    "dev": "next & npm run json-server",
+    "dev": "next & npm run json-graphql-server",
+    "dev:rest": "next & npm run json-server",
     "json-server": "json-server --watch db.json --port 3004",
```
Comment (nathaliebru) [inline package.json:12]: localhost:3333 isn't running when running `npm run dev`, change your `json-server` script to `json-server --watch db.json --port 3004 & json-graphql-server db.json --p 3333`

**Review by nathaliebru — APPROVED** [review]
Looks gooooood!

**Pull request merged**